### PR TITLE
Separated '!' from '?'. Implemented ActorRef stop

### PR DESCRIPTION
**Pull request by mtsokol**
@softinio @mijicd 
Hi!
So here's next part of implementation I wanted to do. In this PR:

- Separated `ask` (`?`) from `tell` (`!`) where `tell` is finally fully operational - caller when sending a message as `tell` is only blocked until recipient's mailbox confirms receiving it (which couldn't be achieved with only a `fork` on previous `!`)
- Exposed actors' `stop` for `Context` in `ActorRef` so actors can be stopped together with all their children. 
- Updated ActorSystem method names according to @mijicd suggestions 
- Bumped `zio` and `zio-nio` version.

Soon I will deliver the next part.

What do you think about it?

### CODE REVIEW

**Comment by mtsokol** [comment]
@mijicd Hi! 
I've just force-pushed some minor tweaks: Changed return type for `make`, bumped fmt version and applied new formatting.

**Comment by mijicd** [comment]
@mtsokol I'll take a look by tomorrow!

File: docs/overview/index.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+title: "Contents"
+---
+
+ZIO Actors - a high-performance, purely-functional library for building, composing, and supervising typed actors backed by `ZIO`.
+Here's list of contents available:
+
+ - **[Basics](basics.md)** — Instantiating `ActorSystem`, defining actor's behavior, spawning actors.
+ - **[Supervision](supervision.md)** — Short description of supervision functionality usage
+ - **[Remoting](remoting.md)** — Defining remoting configuration, usage example, restrictions
+ - **[Queues](queue.md)** — **Not implemented** Support for different types of queues within a mailbox
```
Comment (mijicd) [inline docs/overview/index.md]: I would remove the *not implemented* stuff, and make sure they're reflected in the appropriate issues. Keep in mind the premise - "contents available".

Comment (mtsokol) [reply docs/overview/index.md]: Sure, removed. As I checked they're covered in issues.

**Comment by mijicd** [comment]
Looks good to me, @softinio ?

**Comment by softinio** [comment]
@mijicd Thanks for reviewing dude
@mtsokol Awesome work sir. Just need to get the CI to pass then we can merge. I can take a look this weekend at any CI issues.

Sorry for the delay taking a look. Had a low bandwidth week to work on side projects. Hoping to have more time this weekend.

**Comment by mtsokol** [comment]
@softinio Thanks so much! 
PR is completed since few days ago. This CI failure from yesterday was me accidentally pushing something on my branch 🤦‍♂ - just reverted it, so it's the same final version as before.

**Pull request merged**